samples/streamFileUpload.js

```diff
@@ -0,0 +1,70 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * This application demonstrates how to perform basic operations on files with
+ * the Google Cloud Storage API.
+ *
+ * For more information, see the README.md under /storage and the documentation
+ * at https://cloud.google.com/storage/docs.
+ */
+
+function main(
+  bucketName = 'my-bucket',
+  filePath = './local/path/to/file.txt',
+  destFileName = 'file.txt'
+) {
+  // [START storage_stream_file_upload]
+  /**
+   * TODO(developer): Uncomment the following lines before running the sample
+   */
+  // The ID of your GCS bucket
+  // const bucketName = 'your-unique-bucket-name';
+
+  // The path to your file to upload
+  // const filePath = 'path/to/your/file';
+
+  // The new ID for your GCS file
+  // const destFileName = 'your-new-file-name';
+
+  // Imports the Google Cloud client library
+  const {Storage} = require('@google-cloud/storage');
+
+  // Import Node.js filesystem
+  const fs = require('fs');
+
+  // Creates a client
+  const storage = new Storage();
+
+  // Get a reference to the bucket
+  const myBucket = storage.bucket(bucketName);
+
+  // Create a reference to a file object
+  const file = myBucket.file(destFileName);
```

Comment:
Similarly to my comment on the download sample-- I think the point of this page is that you don't have to use a file for streaming. Maybe we can create a generic stream and show how it could be used to upload from in-memory buffer, or user input, etc? I think this would be more useful and interesting.

Reply:
If I understand the concern, instead of reading from a file the preference would be to read from just any other type of stream? What about if the example showed creating a readable stream from a string and used that to stream to storage?

Reply:
While I agree that showing those things are cool, I think reading a file is a core use case and should be the case shown in the main sample. If customers want other examples, we have those available in our documentation https://googleapis.dev/nodejs/storage/latest/File.html#createWriteStream

Reply:
The c++ sample that already exists uploads from text https://cloud.google.com/storage/docs/streaming#storage-stream-upload-object-cpp
Maybe if we don't use a file we can use text instead to match c++

Reply:
Discussed with @shaffeeullah offline-- I think copying C++ and writing from a readable stream from a string is fine. Let's make sure we have a comment to make it clear that the writes to the stream will continue to be sent until it is closed.